clean up some components

corakingdon opened the pull request:
@davidanthoff I made some changes to some components here which do not affect the base run of the model, but which help when running it in certain extreme scenarios. Do you think it's ok that I've added `max(var, 0)` around certain variables in the neteconomy component which should never be negative? This was causing problems in the welfare component on some extreme runs.

corakingdon committed:
clean up some components

davidanthoff commented:
Hm, I think these `max` terms are tricky... I think in general we shouldn't try to use them to "hide" real problems somewhere else in the model, so I think if we ever add them, we should really do it based on an understand of why.

In thise case, I think the root bound you want to add is that damages can't be larger than GDP, right? So maybe just one limit that enforces that would do the trick? But that is a deviation from the original, and a substantial one... So maybe we should add a flag (i.e. a bool parameter) to the component that can turn that bound on or off, depending on whether one wants to run in Nordhaus original mode or not.

corakingdon committed:
add optional damages and cost caps in neteconomy

corakingdon committed:
add and reorganize tests

corakingdon commented:
@davidanthoff how does this look? I added two optional flags to the neteconomy component. (`cap_damages` and `cap_abatecost`, I'm happy to change the name and/or combine them into only one flag if you think that's better.)

I like the way I've done it here, where we still allow the `DAMAGES` and `ABATECOST` variables to be calculated the same way as the original code, but then if the flags are used, I ensure that `YNET` and `Y` can't be negative, which ensures consumption won't be negative. (Consumption ending up negative is what causes an error in the welfare component.) I'm curious though if you would prefer the actual values of `DAMAGES` and `ABATECOST` to be limited to the size of the economy instead.

I also reorganized the tests and added an additional test file, "test_extremes.jl" which shows the use of these two added flags. Let me know what you think!

corakingdon commented:
two options:
1. add an explicit bound on consumption
2. OR look at older Nordhaus formulation of utility (@davidanthoff will try to find this)

corakingdon commented:
just confirmed that in my current configuration (examples in test_extremes.jl), we do indeed get "-Inf" for UTILITY when consumption goes to zero, even though it doesn't error anymore

corakingdon committed:
clean up some discounting parameters

AlexandrePavlov committed:
Merge branch 'master' into cleaning
